automatic_cross_compile.sh

```diff
@@ -0,0 +1,61 @@
+# This script is intended to be used by the Jenkins night builds.
+# But it still possible to use manually setting up this variables:
+#
+# TARGET=[raspbian]
+# ROS2_DISTRO=[ardent,bouncy,crystal,master,...]
+
+# Check that all variables are defined before start
+if [[ -z "$TARGET" || -z "$ROS2_DISTRO" ]]
+then
+  echo "Error: environment variables are not defined!"
+  echo "Set values to: TARGET / ROS2_DISTRO"
+  exit 1
+fi
+
+THIS_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )"
+
+# Ensure that the cross-compilation environment is built and up-to-date
+BUILD_SCRIPT=$THIS_DIR/build.sh
+bash $BUILD_SCRIPT
+
+# Prepare cross-compilation environment
+SOURCE_SCRIPT=$THIS_DIR/env.sh
+source $SOURCE_SCRIPT $TARGET
+
+# Get sysroot
+SYSROOT_SCRIPT=$THIS_DIR/get_sysroot.sh
+bash $SYSROOT_SCRIPT
```

Comment:
Shouldn't this depend on the TARGET ?

Reply:
The script gets the TARGET value from the corresponding environment variable.
I am still unsure whether to use only env variables or arguments